### PR TITLE
style(DB): battleground_template entries now commented by type (arena…

### PR DESCRIPTION
**Pull request by BarbzYHOOL**
… or bg)

## CHANGES PROPOSED:
- Increased comment length from 32 to 38
- Fixed typo
- Added full name for Nagrand Arena

Can be useful for people who don't know WOW by heart
Can be used to select only arenas or only battlegrounds

## TESTS PERFORMED:
- Applied to DB

## HOW TO TEST THE CHANGES:

 no test needed

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1582810463583691504.sql [hunk 1]
```diff
@@ -0,0 +1,18 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1582810463583691504');
+
+ALTER TABLE `battleground_template`
+CHANGE `Comment` `Comment` char(38) COLLATE 'utf8_general_ci' NOT NULL AFTER `ScriptName`;
+
+UPDATE `battleground_template` SET `Comment` = 'Alterac Valley (battleground)' WHERE `ID` = '1';
+UPDATE `battleground_template` SET `Comment` = 'Warsong Gulch (battleground)' WHERE `ID` = '2';
+UPDATE `battleground_template` SET `Comment` = 'Arathi Basin (battleground)' WHERE `ID` = '3';
+UPDATE `battleground_template` SET `Comment` = 'Nagrand Arena / Ring of Trials (arena)' WHERE `ID` = '4';
```
Comment (Helias) [inline data/sql/updates/pending_db_world/rev_1582810463583691504.sql]: These data are taken from BattlemasterList.dbc, and ID 4 is "Nagrand Arena".

**Comment by Helias** [comment]
I partially agree with this PR because I think that we should make compliant this table with BattlemasterList.dbc where there is a field `InstanceType` which shows you if a row is Battleground or Arena.

For now, we can just merge this but in the future, we should be closed to the DBC content files imho.

File: data/sql/updates/pending_db_world/rev_1582810463583691504.sql (hunk 1 above)
Comment (BarbzYHOOL) [reply data/sql/updates/pending_db_world/rev_1582810463583691504.sql]: https://wow.gamepedia.com/Ring_of_Trials it's just a comment meant to inform the player, when i researched it i couldn't find "nagrand arena"

**Comment by BarbzYHOOL** [comment]
> I partially agree with this PR because I think that we should make compliant this table with BattlemasterList.dbc where there is a field `InstanceType` which shows you if a row is Battleground or Arena.
> 
> For now, we can just merge this but in the future, we should be closed to the DBC content files imho.

you mean there should be an extra row for InstanceType??

**Comment by Helias** [comment]
Yep, because in the DBC there is,
https://wowdev.wiki/DB/BattlemasterList

**Comment by BarbzYHOOL** [comment]
yes but values are 3 and 4, confusing

dunno, if someone wants to voice his opinion

File: data/sql/updates/pending_db_world/rev_1582810463583691504.sql (hunk 1 above)
Comment (BarbzYHOOL) [reply data/sql/updates/pending_db_world/rev_1582810463583691504.sql]: so i feel like my change is better xD

Comment (Helias) [reply data/sql/updates/pending_db_world/rev_1582810463583691504.sql]: could be, but I think that in part we should follow the blizz standard

Comment (BarbzYHOOL) [reply data/sql/updates/pending_db_world/rev_1582810463583691504.sql]: it's just a comment, and in the DBC it's the column "name" not "comment"

**Comment by gengarshadowball** [comment]
In my opinion both of you are right!
1. We should add a comment that allows better understanding of what the battleground template is
2. We should implement hooks and a new column but instead of an integer a bit and update the wiki and add hooks to get this instance Type where 0 is battleground and 1 is arena.

3 and 4 feels like they had other ideas for numbers 0, 1 and 2. Since there is only 2 types of battlegrounds, a bit is small and easy to understand.

**Comment by gengarshadowball** [comment]
All I cans say is if you decide to merge this, I'll add the instanceType to the core of you guys agree.

**Comment by Helias** [comment]
@gengarshadowball I agree with you, you can take the data from DBC and make a parse of the value in this way (like you said):
- 3 => 0 (Battleground)
- 4 => 1 (Arena)

You can call the field "isArena", at the same time, if you have some more time, you can change the function "BattlegroundMgr::isArena()" to check the DBC data (imported to the database, so check the database) instead of using a custom algorithm.

This is the algorithm which annoys me.
image

**Comment by BarbzYHOOL** [comment]
@gengarshadowball ;)

**Pull request merged**